### PR TITLE
Add state methods to delegate

### PR DESCRIPTION
**Pull request by 0xch4z**

### CODE REVIEW

File: DrawerController/DrawerController.swift [hunk 1]
```diff
@@ -305,6 +314,8 @@ open class DrawerController: UIViewController, UIGestureRecognizerDelegate {
         /// Golden-Path
         guard isEnable() else { return }
         guard !isAnimating else { return }
+
+        delegate?.drawerWillCloseSide?(drawerController: self)
```
Comment (kawoou) [inline DrawerController/DrawerController.swift]: Could you pass this state using the `self.drawerSide` value?

**Comment by kawoou** [comment]
Please support Rx delegating.

**Comment by 0xch4z** [comment]
Ah, forgot about Rx. Just pushed it

File: DrawerController/DrawerController.swift [hunk 2]
```diff
@@ -358,7 +358,7 @@ open class DrawerController: UIViewController, UIGestureRecognizerDelegate {
                     ss.didFinishAnimate(side: .none, percent: 0.0)
                     
                     ss.isAnimating = false
-                    ss.delegate?.drawerDidCloseSide?(drawerController: ss)
+                    ss.delegate?.drawerDidCloseSide?(drawerController: ss, side: ss.drawerSide)
```
Comment (kawoou) [inline DrawerController/DrawerController.swift]: If you do not write oldValue, there is likely to be some variation in the middle.

```swift
let oldDrawerSide = drawerSide

delegate?.drawerWillCloseSide?(drawerController: self, side: oldDrawerSide)

...

ss.delegate?.drawerDidCloseSide?(drawerController: ss, side: oldDrawerSide)

...
```

File: DrawerController/Rx/DrawerController+Rx.swift [hunk 3]
```diff
@@ -125,5 +125,32 @@ extension Reactive where Base: DrawerController {
...
+        return ControlEvent(events: source)
+    }
+    public var didOpenSide: ControlEvent<DrawerSide> {
+        let source = delegate
+            .methodInvoked(#selector(DrawerControllerDelegate.drawerDidOpenSide(drawerController:side:)))
+            .map { try castOrThrow(DrawerSide.self, $0[1]) }
+        
+        return ControlEvent(events: source)
+    }
+    public var didCloseSide: ControlEvent<DrawerSide> {
```
Comment (kawoou) [inline DrawerController/Rx/DrawerController+Rx.swift:149]: 👍

**Comment by 0xch4z** [comment]
Updated

**Review by kawoou — APPROVED** [review]
Thank you!
LGTM

**Comment by kawoou** [comment]
Merged

**Pull request merged**